app/src/main/java/ch/epfl/sdp/peakar/points/POICache.java

```diff
@@ -0,0 +1,155 @@
+        Gson gson = new Gson();
+        //Convert to JSON
+        String serializesCache = gson.toJson(poiCacheContent);
+        saveJson(serializesCache,context);
+    }
+
+    /**
+     * Save the serialized JSON to text file
+     * @param serializedObject string in json format
+     * @param context context
+     */
+    private static void saveJson(String serializedObject,Context context){
+        File cacheDir = context.getCacheDir();
+        File outputFile = new File(cacheDir,CACHE_FILE_NAME);
+        try{
+            FileWriter writer = new FileWriter(outputFile);
+            writer.append(serializedObject);
+            writer.flush();
+            writer.close();
+        }
+        catch(IOException e){
+            Log.e("Exception", "File write failed cache POIPoints: " + e.toString());
+        }
+    }
+
+
+    /**
+     * Read text file and retrieve serialized
+     * @return serialized string
+     */
+    private static String readJSON(Context context){
+        File file = new File(context.getCacheDir(),CACHE_FILE_NAME);
+        StringBuilder fileContent = new StringBuilder();
+        try{
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+            String line;
+            while ((line = bufferedReader.readLine()) != null){
+                fileContent.append(line);
+                fileContent.append('\n');
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return fileContent.toString();
+    }
+
+    /**
+     * Retrieve POI data from cache and overwrite cachedPOIPoints and cachedBoundingBox
+     * with cache file values
+     */
+    private static void retrievePOIDataFromCache(Context context){
+        String fileContent = readJSON(context);
+        Gson gson = new Gson();
+        POICacheContent poiCacheContent = gson.fromJson(fileContent, POICacheContent.class);
+        cachedPOIPoints = poiCacheContent.getCachedPOIPoints();
+        cachedBoundingBox = poiCacheContent.getCachedBoundingBox();
+    }
+
+    /**
+     * Check if the user is located inside the middle part of the bounding box
+     * Using cached bounding box, a shrink-ed bounding box is computed.
+     * The methods then checks if the user position is somewhere inside this
+     * smaller bounding box
+     * @param userPoint user location
+     * @return true if the user is in the middle of the cached bounding box
+     *          false if not
+     */
+    public boolean isUserInBoundingBox(UserPoint userPoint,Context context){
+        //Get the bounding box from file if not already present
+        if(cachedBoundingBox == null)
+            retrievePOIDataFromCache(context);
+        //Decrease the size of the bounding box by half
+        //BoundingBox innerBox = cachedBoundingBox.increaseByScale(INNER_BOUNDING_BOX_SCALING_FACTOR);
```

Comment:
This line is commented. Is it an error or was it on purpose? If so, can you add a comment explaining why (for example, if we can use it in future)?

Reply:
Uncommented in next commit